src/components/SearchModal/CurrencySearch.tsx

```diff
@@ -116,14 +117,17 @@ export function CurrencySearch({
   const native = useNativeCurrency()
 
   const filteredSortedTokensWithETH: Currency[] = useMemo(() => {
-    if (!native) return filteredSortedTokens
+    // Use Celo ERC20 Implementation and exclude the native asset
+    if (!native || (chainId && [SupportedChainId.CELO, SupportedChainId.CELO_ALFAJORES].includes(chainId))) {
```

Comment:
Why do you include the ERC20 in the list, instead of using the native asset (which I _think_ will be wrapped in the ERC20 for use in the swap widget, anyways)?

Usually, we list native assets first, unless they are filtered out. Doing it this way means that Celo will always be listed alphabetically (not first, and not even on the initial token selector screen without scrolling).

This feels like an antipattern.

Reply:
resolved with conversation offline + your commit